### PR TITLE
chore: update Storage gRPC protobuf defintitions

### PR DESCRIPTION
**Pull request by e-dard**
Closes https://github.com/influxdata/influxdb_iox/issues/3164

This PR updates the protobuf definitions for interaction with the Storage gRPC API.

There are some notable changes to the gRPC API:

 - I have removed `hints` from `ReadGroupRequest`. It's set to be deprecated and unused. I have reserved the field.
 - ~All the `enum` variants I have set to capitalised `SNAKE_CASE` because of a defect in `pbjson`, which is tracked in https://github.com/influxdata/pbjson/issues/15. Happy to revisit the casing when that's fixed.~
 - The protobuf definition now includes `TagValuesGroupedByMeasurementAndTagKeyRequest`. I have added https://github.com/influxdata/influxdb_iox/issues/3177 to track whether we want to implement support for this.
 - The definition now includes `TagKeyMetaNames`. This is needed as part of the work towards https://github.com/influxdata/influxdb_iox/issues/3166.

### Testing

As well as all the IOx tests passing I have re-ran the internal Flux test suite against IOx and confirmed that this change has not caused any regressions.

### Future Work

I plan to investigate internally how we might unify the Protobuf definitions between projects so that we don't have the burden of maintaining them across projects.

### CODE REVIEW

**Review by mkmik — COMMENTED** [review]
I assume we don't need pbjson for this schema.

Can't we just disable jsonpb generation for this proto package and share (copy verbatim) the same IDL as it's defined in IDPE without having to perform fixups in the sources?

**Comment by e-dard** [comment]
@mkmik 

> Can't we just disable jsonpb generation for this proto package and share (copy verbatim) the same IDL as it's defined in IDPE without having to perform fixups in the sources?

I don't know, can we? :-)

If you know we can please let me know how to do that and I'm happy to update the definitions to use original casing.

**Comment by e-dard** [comment]
Build suddenly decided it's unhappy. Looking into it.

**Comment by mkmik** [comment]
> @mkmik
> 
> > Can't we just disable jsonpb generation for this proto package and share (copy verbatim) the same IDL as it's defined in IDPE without having to perform fixups in the sources?
> 
> I don't know, can we? :-)
> 
> If you know we can please let me know how to do that and I'm happy to update the definitions to use original casing.

```rust
    pbjson_build::Builder::new()
        .register_descriptors(&descriptor_set)?
        .build(&[".influxdata", ".google.longrunning", ".google.rpc"])?;
```

```console
$ ls generated_types/protos/influxdata/
iox  pbdata  platform

I _think_ that we could simply just enable pbjson for the iox stuff:

```rust
    pbjson_build::Builder::new()
        .register_descriptors(&descriptor_set)?
        .build(&[".influxdata.iox", ".google.longrunning", ".google.rpc"])?;
```

which will

File: generated_types/protos/influxdata/platform/storage/storage_common.proto [hunk 1]
```diff
@@ -9,59 +9,93 @@ package influxdata.platform.storage;
...
+// TODO(edd): when https://github.com/influxdata/pbjson/issues/15 is fixed
+// the variants can use camel case again.
+enum TagKeyMetaNames {
+  // option (gogoproto.goproto_enum_prefix) = false;
+
+  // TagKeyMetaNamesText means the tag keys for measurement and field will
+  // be returned as _measurement and _field respectively.
+  TagKeyMetaNamesText = 0;
+
+  // TagKeyMetaNames means the tag keys for measurement and field will
```
Comment (alamb) [inline generated_types/protos/influxdata/platform/storage/storage_common.proto:32]: 👍

File: generated_types/protos/influxdata/platform/storage/storage_common.proto [hunk 2]
```diff
@@ -165,40 +199,55 @@ message TimestampRange {
...
 // TagValuesRequest is the request message for Storage.TagValues.
 message TagValuesRequest {
-  google.protobuf.Any tags_source = 1 ;
+  google.protobuf.Any TagsSource = 1; 
   TimestampRange range = 2; // [(gogoproto.nullable) = false];
   Predicate predicate = 3;
-
-  // string tag_key = 4;
-  // AAL changed from string --> bytes to handle \xff literals in Rust which are not valid UTF-8
   bytes tag_key = 4;
```
Comment (alamb) [inline generated_types/protos/influxdata/platform/storage/storage_common.proto:212]: Interesting -- this got done in IDPE as well

**Review by alamb — APPROVED** [review]
👍  

I took a look through https://github.com/influxdata/idpe/tree/master/storage/storageproto and it looks like this is a pretty faithful migration so 👍  for that

I also noticed there changes to `predicate` in  https://github.com/influxdata/idpe/pull/12368 from @wolffcm  that need to get ported over at some point as well

The changes to rust also look good

**Comment by e-dard** [comment]
@mkmik address `pbjson` comment in https://github.com/influxdata/influxdb_iox/pull/3178/commits/330f6ea7b47c712db0fa62db2d5c093e0d381c37

**Comment by e-dard** [comment]
@alamb regarding that predicate change - I have mentioned the idea that any subsequent changes also include updates to the capabilities call, so that IOx does not need to implement functionality immediately. cc @wolffcm

**Comment by wolffcm** [comment]
@e-dard @alamb For your records, here is the idpe issue to update the capabilities API:
https://github.com/influxdata/idpe/issues/12399
We won't be sending any of the new predicates over gRPC unless the capability is there, so you can update at your leisure (but queries could be slow).

**Pull request merged**